autorest/sender.go

```diff
@@ -222,11 +254,12 @@ func WithLogging(logger *log.Logger) SendDecorator {
 }
 
 // DelayForBackoff invokes time.After for the supplied backoff duration raised to the power of
-// passed attempt (i.e., an exponential backoff delay). Backoff may be zero. The delay may be
-// canceled by closing the passed channel. If terminated early, returns false.
+// passed attempt (i.e., an exponential backoff delay). Backoff duration is in seconds and can set
+// to zero for no delay. The delay may be canceled by closing the passed channel. If terminated early,
+// returns false.
 func DelayForBackoff(backoff time.Duration, attempt int, cancel <-chan struct{}) bool {
 	select {
-	case <-time.After(time.Duration(math.Pow(float64(backoff), float64(attempt)))):
+	case <-time.After(time.Duration(math.Pow(backoff.Seconds(), float64(attempt))) * time.Second):
```

Comment:
fair. I'd implement this with `backoff * (2^attempt)`. I don't think `backoff^attempt` is a very good idea, for 10s, it will go off quite exponentially.